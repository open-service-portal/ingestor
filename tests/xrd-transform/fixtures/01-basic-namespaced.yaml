apiVersion: apiextensions.crossplane.io/v2
kind: CompositeResourceDefinition
metadata:
  name: databases.example.com
  annotations:
    backstage.io/description: "Basic namespaced database provisioning"
    openportal.dev/tags: "database,storage,sql"
spec:
  group: example.com
  names:
    kind: Database
    plural: databases
  scope: Namespaced
  versions:
    - name: v1alpha1
      served: true
      referenceable: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              properties:
                engine:
                  type: string
                  description: Database engine type
                  enum:
                    - postgres
                    - mysql
                    - mongodb
                  default: postgres
                size:
                  type: string
                  description: Database size
                  enum:
                    - small
                    - medium
                    - large
                  default: small
              required:
                - engine
