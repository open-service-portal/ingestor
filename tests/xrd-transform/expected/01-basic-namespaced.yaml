apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
metadata:
  name: databases-example-com
  title: Database
  description: Create and manage Database resources
  tags:
    - crossplane
    - example-com
    - database
    - storage
    - sql
  annotations:
    backstage.io/managed-by: xrd-transform
    backstage.io/managed-by-location: provider:XRDTemplateEntityProvider
    backstage.io/managed-by-origin-location: provider:XRDTemplateEntityProvider
    crossplane.io/xrd-name: databases.example.com
    crossplane.io/xrd-group: example.com
spec:
  owner: platform-team
  type: crossplane-resource
  parameters:
    - title: Resource Configuration
      required:
        - name
        - namespace
        - engine
      properties:
        name:
          title: Name
          type: string
          description: Name of the Database
          pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
        namespace:
          title: Namespace
          type: string
          description: Kubernetes namespace
          default: default
        engine:
          title: Engine
          type: string
          description: Database engine type
          default: null
          enum:
            - null
            - null
            - null
        size:
          title: Size
          type: string
          description: Database size
          default: null
          enum:
            - null
            - null
            - null
  steps:
    - id: workflow-info
      name: Direct Apply Workflow Information
      action: debug:log
      input:
        message: |
          ⚡ This template uses direct apply workflow:

          1. Creates your Database resource immediately in the cluster
          2. Registers the resource in the Backstage catalog

          Your resource will be created as soon as you click Create.
      output:
        text:
          - title: ⚡ Direct Apply Workflow
            content: >
              This template follows a **direct apply workflow** for immediate
              resource creation:


              **What happens when you click Create:**


              1. **Create Resource Immediately** 🚀
                 - Your Database resource is created directly in the Kubernetes cluster
                 - No approval required - resource is active immediately
                 - Applied to cluster: `rancher-desktop`
                 - Created in namespace: `$`

              2. **Register in Catalog** 📝
                 - The resource is automatically registered in the Backstage software catalog
                 - You can track and manage it through Backstage

              **✅ Important**: Your resource is **created immediately** without
              approval workflow.
    - id: create-resource
      name: Create Database
      action: kube:apply
      input:
        manifest: |
          apiVersion: example.com/v1alpha1
          kind: Database
          metadata:
            name: ${{ parameters.name }}
            namespace: ${{ parameters.namespace }}
          spec: ${{ parameters }}
        namespaced: true
      output:
        text:
          - title: Resource Details
            content: |
              **Resource Created Successfully!** ✅

              **Resource Information:**
              - **Name**: `$`
              - **Type**: Database
              - **API Version**: example.com/v1alpha1
              - **Cluster**: rancher-desktop
              - **Namespace**: `$`

              Your resource is now active in the cluster and ready to use.
    - id: register
      name: Register in Catalog
      action: catalog:register
      input:
        repoContentsUrl: ${{ steps.create-resource.output.repoContentsUrl }}
        catalogInfoPath: /catalog-info.yaml
      output:
        links:
          - title: View in Catalog
            icon: catalog
            url: ${{ steps['register'].output.entityRef }}
        text:
          - title: Catalog Registration
            content: |
              **Registered in Backstage Catalog** 📝

              Your Database resource has been added to the software catalog.
              You can now track, monitor, and manage it through Backstage.

